app/Http/Controllers/Api/SignupsController.php

```diff
@@ -50,12 +50,18 @@ public function store(SignupRequest $request)
     {
         $transactionId = incrementTransactionId($request);
 
-        $signup = $this->signups->create($request->all(), $transactionId);
+        // Check to see if the signup exists before creating one.
+        $signup = $this->signups->get($request['northstar_id'], $request['campaign_id'], $request['campaign_run_id']);
 
-        if ($signup) {
+        if (! $signup) {
+            $signup = $this->signups->create($request->all(), $transactionId);
             $code = 201;
         }
 
+        if ($signup) {
+            $code = 200;
```

Comment:
Will this conditional always run, since `$signup` is created on line 57 if it didn't already exist?

It might be easier to set the response code before we try to make the new signup:

```php
$signup = $this->signups->get(...);
$code = $signup ? 200 : 201;

if (! $signup) {
    $signup = // ...
```